### ISSUE TITLE
Custom Container View Controller

### ISSUE THREAD

**Issue by marcelofabri** [uacaps/PageMenu#2]
I've found this component's UX very well designed. However, it does not correctly implements a custom container view controller. This is important to get right because of the lifecycle of child view controllers.

There's more info about this here: https://developer.apple.com/library/ios/featuredarticles/ViewControllerPGforiPhoneOS/CreatingCustomContainerViewControllers/CreatingCustomContainerViewControllers.html

**Comment by fahlout** [comment]
Thanks for your feedback. I'll look more into it in the morning and will look into fixing this issue as soon as possible.

**Comment by fahlout** [comment]
I have added the view controllers as child view controllers as explained in the link you sent. I'm not sure what else needs to be added/changed within my library. Can you lead me in the right direction if there is more involved in it than what I changed so far? Thanks! I'm definitely looking to conform to the guidelines and get the life cycle of each view controller right.

**Comment by marcelofabri** [comment]
It all depends on when one expects `viewDidAppear` & friends to be called. IMO, they should be called when they are actually selected (and not on the menu's creation). To do so, a bit more of working needs to be done. I'm pasting an implementation I have for a project (in Objective-C). 

This is the method called to show a view controller on my custom container:

```objc
- (void)showViewControllerAtIndex:(NSInteger)idx
    if (idx == self.viewControllerIndex) return;
    
    if (self.viewControllerIndex >= 0) {
        UIViewController *old = self.viewControllers[self.viewControllerIndex];
        [old willMoveToParentViewController:nil];
        
        [old.view removeFromSuperview];
        [old removeFromParentViewController];
        
        [old didMoveToParentViewController:nil];
    }

    
    UIViewController *vc = self.viewControllers[idx];
    
    [vc willMoveToParentViewController:self];

   // not using auto layout here
    vc.view.frame = self.containerView.bounds;
    
    // make sure that it resizes on rotation automatically
    vc.view.autoresizingMask = self.containerView.autoresizingMask;
    
    [self addChildViewController:vc];
    [self.containerView addSubview:vc.view];
    [vc didMoveToParentViewController:self];
    
    self.viewControllerIndex = idx;
```

**Comment by fahlout** [comment]
I am close to releasing an updated version of this library. I just have one question I would like to get your feedback on. Lets say I have 3 menu items (Menu1, Menu2, Menu3) and Menu1 is currently selected. If I then click on Menu3 it will call viewWillAppear on Menu2 and then on Menu3 because it needs to move through Menu2 to get to Menu3. Do you think that is an acceptable behavior? Thanks for your time!

**Comment by marcelofabri** [comment]
Yeah, I think it's ok. I'd only just add a note about this somewhere.

**Comment by fahlout** [comment]
Resolved in version 1.1.0
